ec2uploadimg

```diff
@@ -43,8 +52,12 @@ def print_ex(msg=None):
         logger.info("{}: {}".format(exc_type.__name__, value))
 
 
+# ----------------------------------------------------------------------------
 def signal_handler(signum, frame):
+    """Handle signals and clean up resources properly"""
     global aborted
+    global uploader
+    global setup
```

Comment:
There are no changes in value to `uploader` or `setup` in this function, as such they do not need to be declared as `global` Python will find the names in the namespace lookup progression from function -> module -> other enclosing name spaces.

`aborted` has to be declared `global` because it's value is changed in the function and if not declared as `global` Python would only create a local assignment and the module variable value would not change.

Reply:
Done!